NOTICE.txt

```diff
@@ -0,0 +1,3968 @@
+
+
+Files covered:
+
+spyder/widgets/waitingspinner.py
+
+
+-------------------------------------------------------------------------------
+
+
+
+DataFrameModel from QtPandas (Pandas 0.13.1)
+--------------------------------------------
+
+
+Copyright (c) 2008-2011 AQR Capital Management, LLC
+Copyright (c) 2011-2012 Lambda Foundry, Inc. and PyData Development Team
+Copyright (c) 2013 Jev Kuznetsov and contributors
+Copyright (c) 2014- Spyder Project Contributors
+
+
+Author: Jev Kuznetsov and pandas contributors | https://github.com/sjev
+Site: https://pandas.pydata.org/
+Source: https://github.com/draperjames/qtpandas
+License: BSD (3-clause) License | https://opensource.org/licenses/BSD-3-Clause
+
+Extensive adaptation, modification and improvements made to the base class.
+
+
+pandas is distributed under a 3-clause ("New") BSD license.
+
+AQR Capital Management began pandas development in 2008. Development was
+led by Wes McKinney. AQR released the source under this license in 2009.
+Wes is now an employee of Lambda Foundry, and remains the pandas project
+lead.
+
+The PyData Development Team is the collection of developers of the PyData
+project. This includes all of the PyData sub-projects, including pandas. The
+core team that coordinates development on GitHub can be found here:
+https://github.com/pydata.
+
+Full credits for pandas contributors can be found in the documentation.
+
+PyData uses a shared copyright model. Each contributor maintains copyright
+over their contributions to PyData. However, it is important to note that
+these contributions are typically only changes to the repositories. Thus,
+the PyData source code, in its entirety, is not the copyright of any single
+person or institution. Instead, it is the collective copyright of the
+entire PyData Development Team. If individual contributors want to maintain
+a record of what changes/contributions they have specific copyright on,
+they should indicate their copyright in the commit message of the change
+when they commit the change to one of the PyData repositories.
+
+
+See below for the full text of the BSD (3-clause) license.
+
+The current pandas license can be viewed at:
+https://github.com/pandas-dev/pandas/blob/master/LICENSE
+
+The current QtPandas license can be viewed at:
+https://github.com/draperjames/qtpandas/blob/master/LICENSE
+
+The original version of the original file can be viewed at:
+https://github.com/pandas-dev/pandas/blob/d10a658673b7d2c2e7a346461c9a4bfc5233d7e1/pandas/sandbox/qtpandas.py
+
+The current version of the original file can be viewed at:
+https://github.com/draperjames/qtpandas/blob/master/qtpandas/models/DataFrameModel.py
+
+
+Files covered:
+
+spyder/widgets/variableexplorer/dataframeeditor.py
+
+
+===============================================================================
+
+
+
+
+External Fonts Used for Icons
+=============================
+
+
+Font Awesome 4.7 (via QtAwesome)
```

Comment:
This makes no sense. We don't distribute FontAwesome. Why do we need this here?

Reply:
This notice needs to be moved to QtAwesome. Please remove it from here.

Reply:
> This makes no sense. We don't distribute FontAwesome. Why do we need this here?

Good question, and one on which I initially came to the same conclusion as you—I didn't want to "open that can of worms" for the same reasons you probably don't. However, on closer inspection there were very particular reasons why FA is the one exception to not listing anything that isn't physically included in the Spyder repo, reasons which make much more sense if one considers that "distribution" is not always synonymous with what actually matters under copyright law and ethics thereof—that is, whether the original work is a meaningful part of the end product and its presentation.

Even if not physically included in the package, they comprise the vast majority of the standard Spyder iconset that is actively displayed as part of Spyder's GUI itself (rather than as output, data, or reprisentations thereof) to the end user, forming an integral piece in the user-visible end-product that is Spyder, as much if not more so than any other icon that is included. If they weren't both graphical, an icon font (rather than a typographical one) and made a part of Spyder's core graphical UI (rather than e.g. as part of user output), then we could get away with it, but as it stands they deserve the same treatment as any other Spyder icon, as they are as much a part as anything of the visual pastiche that is Spyder.

Is it still somewhat gray? Yes, it's not 100% certain the courts would rule in their favor, but given there's some very clear distinctions with this specific case that prevent it from becoming a slippery slope and applying to any (other) Spyder dependencies (and recall, it is specific to Font Awesome the asset, not QtAwesome the dependency—I don't list the details for that) it costs us little to nothing to add it, shouldn't require any further maintenance, and is unlikely at best to apply to anything in the future (and even if arguably so, doesn't obligate us to do so).

> This notice needs to be moved to QtAwesome

Indeed it does; it certainly needs to go there as well.

Reply:
It also has the benefit of clearly documenting, in the same consistent place, exactly where all of Spyder's primary iconset is derived from, for both users and developers interested in finding out more (or even using QtAwesome). Otherwise, one has to dig through the code to find out exactly where they're originally from, and even then it may not be 100%. So again, even if this were somehow "out of date", still better to have something than nothing as a useful starting point for those curious.